papas/pfobjects.py

```diff
@@ -74,11 +62,13 @@ def __init__(self, energy, position, size_m, layer='ecal_in',index=0, particle=N
         if identifiervalue== None:
             identifiervalue = max(energy, 0.)
         if layer == 'ecal_in':
-            super(Cluster, self).__init__(Identifier.PFOBJECTTYPE.ECALCLUSTER, index, self.subtype, identifiervalue)
+            idtype = Identifier.PFOBJECTTYPE.ECALCLUSTER
         elif layer == 'hcal_in':
-            super(Cluster, self).__init__(Identifier.PFOBJECTTYPE.HCALCLUSTER, index, self.subtype, identifiervalue)
+            idtype = Identifier.PFOBJECTTYPE.HCALCLUSTER
         else :
             assert (False)
+        self.uniqueid = Identifier.make_id(idtype, index, self.subtype, identifiervalue)  
+        super(Cluster, self).__init__()
```

Comment:
why removing the parameters of the contructor of the base class?

Reply:
I think we need to discuss!